src/transformState.js

```diff
@@ -5,7 +5,25 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const object of actions) {
+    if (object.type === 'removeProperties') {
+      for (const values of object.keysToRemove) {
+        delete state[values];
+      }
+    }
+
+    if (object.type === 'addProperties') {
+      for (const pair in object.extraData) {
```

Comment:
You can use Object.assign() instead of this loop and add obj.extraData into state.